manta-trusted-setup/src/groth16/bn254/manta_pay.rs

```diff
@@ -0,0 +1,363 @@
+            .serialize(&mut hasher)
+            .expect("Consuming ratio proof of tau failed.");
+        proof
+            .alpha
+            .serialize(&mut hasher)
+            .expect("Consuming ratio proof of alpha failed.");
+        proof
+            .beta
+            .serialize(&mut hasher)
+            .expect("Consuming ratio proof of beta failed.");
+        into_array_unchecked(hasher.0.finalize())
+    }
+}
+
+/// An accumulator for phase 1 parameters whose size is sufficient
+/// for all MantaPay circuits.
+pub type MantaPayAccumulator = Accumulator<MantaPaySetupCeremony>;
+
+impl MpcConfiguration for MantaPaySetupCeremony {
+    type Challenge = [u8; 64];
+    type Hasher = BlakeHasher;
+
+    #[inline]
+    fn challenge(
+        challenge: &Self::Challenge,
+        prev: &MpcState<Self>,
+        next: &MpcState<Self>,
+        proof: &MpcProof<Self>,
+    ) -> Self::Challenge {
+        let mut hasher = Self::Hasher::default();
+        hasher.0.update(challenge);
+        prev.serialize(&mut hasher)
+            .expect("Consuming the previous state failed.");
+        next.serialize(&mut hasher)
+            .expect("Consuming the current state failed.");
+        proof
+            .serialize(&mut hasher)
+            .expect("Consuming proof failed");
+        into_array_unchecked(hasher.0.finalize())
+    }
+}
+
+impl Types for MantaPaySetupCeremony {
+    type State = MpcState<Self>;
+    type Challenge = [u8; 64];
+    type Proof = MpcProof<Self>;
+}
+
+impl ProvingKeyHasher<Self> for MantaPaySetupCeremony {
+    type Output = [u8; 64];
+
+    #[inline]
+    fn hash(proving_key: &ProvingKey<<Self as Pairing>::Pairing>) -> Self::Output {
+        let mut hasher = BlakeHasher::default();
+        proving_key
+            .serialize(&mut hasher)
+            .expect("Hasher is not allowed to fail");
+        into_array_unchecked(hasher.0.finalize())
+    }
+}
+
+// impl<P> Serializer<short_weierstrass_jacobian::GroupAffine<P>> for MantaPaySetupCeremony
+// where P: SWModelParameters {
+//     fn serialize_unchecked<W>(item: &short_weierstrass_jacobian::GroupAffine<P>, writer: &mut W) -> Result<(), std::io::Error>
+//     where
+//         W: Write {
+//         CanonicalSerialize::serialize_unchecked(item, writer).into() // TODO: How to convert to std::io::Error ?
+//     }
+
+//     fn serialize_uncompressed<W>(item: &short_weierstrass_jacobian::GroupAffine<P>, writer: &mut W) -> Result<(), std::io::Error>
+//     where
+//         W: Write {
+//         todo!()
+//     }
+
+//     fn uncompressed_size(item: &short_weierstrass_jacobian::GroupAffine<P>) -> usize {
+//         todo!()
+//     }
+
+//     fn serialize_compressed<W>(item: &short_weierstrass_jacobian::GroupAffine<P>, writer: &mut W) -> Result<(), std::io::Error>
+//     where
+//         W: Write {
+//         todo!()
+//     }
+
+//     fn compressed_size(item: &short_weierstrass_jacobian::GroupAffine<P>) -> usize {
+//         todo!()
+//     }
```

Comment:
Here I just want to use its `CanonicalSerialize` implementation but I can't figure out how to convert errors